content/docs/rfcs/17/README.md

```diff
@@ -2,127 +2,136 @@
 slug: 17
 title: 17/WAKU-RLN-RELAY
 name: Waku v2 RLN Relay
-status: raw
+status: draft
 tags: waku-core
 editor: Sanaz Taheri <sanaz@status.im>
 ---
 
-The current specification embodies the details of the spam-protected version of `relay` protocol empowered by Rate Limiting Nullifiers (RLN). 
-<!-- More details on RLN can be found in [this spec]() (TODO: to link the spec). -->
+The `17/WAKU-RLN-RELAY` protocol is an extension of `11/WAKU-RELAY` which additionally provides spam protection using [Rate Limiting Nullifiers (RLN)](/spec/32). 
 
-The security objective is to control the number of PubSub messages that each peer can publish per epoch where epoch is a system design parameter, regardless of the published topic.
+The security objective is to contain spam activity in a GossipSub network by enforcing a global messaging rate to all the peers.
+Peers that violate the messaging rate are considered spammers and their message is considered spam.
+Spammers are also financially punished and removed from the system. 
 
 
 **Protocol identifier***: `/vac/waku/waku-rln-relay/2.0.0-alpha1`
 
 # Motivation
 
-In open p2p messaging networks, one big problem is spam-resistance. 
-Existing solutions, such as Whisper’s proof of work, are insufficient, especially for heterogeneous nodes. 
+In open and anonymous p2p messaging networks, one big problem is spam resistance. 
+Existing solutions, such as Whisper’s proof of work are computationally expensive hence not suitable for resource-limited nodes. 
 Other reputation-based approaches might not be desirable, due to issues around arbitrary exclusion and privacy.
 
-We augment the `relay` protocol with a novel, light, and effective spam prevention mechanism which also suits the resource-constrained nodes.
-
-<!-- TODO: Fill in more -->
+We augment the [`11/WAKU2-RELAY`](/spec/11) protocol with a novel construct of [RLN](/spec/32) to enable an efficient economic spam prevention mechanism that can be run in resource-constrained environments.
 
 
 # Flow
+The messaging rate is set to 1 per `epoch`, 
+where `epoch` indicates the number of `T` seconds elapsed since the Unix epoch event.
+The value of `T` is application-dependent.
+See section [Recommended System Parameters](#recommended-system-parameters) for some recommended ways to set a sensible `T` value.
+Peers subscribed to a spam-protected `pubsubTopic` are only allowed to send one message per `epoch`
+(messaging rate MAY be enforced for `WakuMessages` with a specific `contentTopic` published on a `pubsubTopic`, but that is an implementation choice).
```

Comment:
What do we mean by an implementation choice?

Perhaps we could add this as a new paragraph, remove () and say "could be" but "not supported right now ~"

Reply:
Sounds like it is underspecified right now? Which is ok, we can make it explicit. Implementation choice makes it seem arbitrary, but perhaps it is more that this is deferred to a higher level layer, such a specific chat protocol built on top?

Reply:
Right, it is up to the higher-level protocols to dictate such granularity. I'll update the description accordingly.